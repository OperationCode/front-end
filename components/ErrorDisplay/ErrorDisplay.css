.ErrorDisplay {
  background: #121212;

  /* TODO: calc min-height based upon 100vh - nav and footer sizes */
  height: 100vh;
  width: 100vw;
}

.ErrorDisplay .bg {
  align-items: center;
  background-image: url("/static/images/TankFlip.gif");
  background-size: cover;
  color: white;
  display: flex;
  flex-direction: column;

  /* TODO: calc min-height based upon 100vh - nav and footer sizes */
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.title {
  font-size: 4rem;
}

.paragraph {
  font-size: 1.4rem;
}
