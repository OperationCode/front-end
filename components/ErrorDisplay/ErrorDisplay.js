import React, { useEffect } from 'react';
import { number } from 'prop-types';
import Head from 'components/head';
import styles from './ErrorDisplay.module.css';

ErrorDisplay.propTypes = { statusCode: number };

ErrorDisplay.defaultProps = { statusCode: undefined };

function ErrorDisplay({ statusCode }) {
  useEffect(() => {
    document.body.style.overflowY = 'hidden';
    return () => {
      document.body.style.overflowY = '';
    };
  }, []);

  return (
    <>
      <Head title={statusCode ? `${statusCode}` : 'Error'}>
        <meta name="robots" content="noindex, nofollow" key={statusCode} />
      </Head>

      <div className={styles.ErrorDisplay}>
        <div className={styles.bg}>
          <div className={styles.textContainer}>
            <h1 className={styles.title}>{statusCode || 'Oh no'}!</h1>
            <p className={styles.paragraph}>
              We&apos;re so ashamed. You definitely weren&apos;t supposed to see this...
            </p>
          </div>
        </div>
      </div>
    </>
  );
}

export default ErrorDisplay;
