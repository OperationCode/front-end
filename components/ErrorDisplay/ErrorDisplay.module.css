.ErrorDisplay {
  background: #121212;
  height: 100%;
  width: 100%;
}

.ErrorDisplay .bg {
  align-items: center;
  background-image: url("/static/images/TankFlip.gif");
  background-size: cover;
  background-position-x: center;
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;
  height: calc(100vh - 390px);
}

@media screen and (--medium-viewport) {
  .ErrorDisplay .bg {
    height: 100vh;
  }
}

.textContainer {
  text-align: center;
  margin: 1rem auto;
}

.title {
  font-size: 4rem;
}

.paragraph {
  font-size: 1.4rem;
}
