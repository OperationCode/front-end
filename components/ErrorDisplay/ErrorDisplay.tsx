import Head from 'components/head';

export interface ErrorDisplayPropsType {
  /**
   * Displasy a status code instead of 'Error'.
   */
  statusCode?: number;
}

function ErrorDisplay({ statusCode }: ErrorDisplayPropsType) {
  return (
    <>
      <Head title={statusCode ? `${statusCode}` : 'Error'}>
        <meta name="robots" content="noindex, nofollow" key={statusCode} />
      </Head>

      <div
        className={
          'h-screen text-white bg-[url("/static/images/TankFlip.gif")] flex flex-col justify-center w-full  bg-cover items-center'
        }
      >
        <div className="mx-auto my-4 text-center">
          <h1 className="text-6xl text-white">{statusCode || 'Oh no'}!</h1>
          <p className="text-2xl text-white">
            We&apos;re so ashamed. You definitely weren&apos;t supposed to see this...
          </p>
        </div>
      </div>
    </>
  );
}

export default ErrorDisplay;
