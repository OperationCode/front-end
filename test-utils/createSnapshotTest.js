/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
