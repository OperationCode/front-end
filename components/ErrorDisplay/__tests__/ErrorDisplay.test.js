import { render } from '@testing-library/react';
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';
import ErrorDisplay from '../ErrorDisplay';

describe('ErrorDisplay', () => {
  it('should render with just required props', () => {
    createSnapshotTest(<ErrorDisplay statusCode={404} />);
  });

  it('should render h1, even when no statusCode is passed', () => {
    const { container } = render(<ErrorDisplay />);
    expect(container.querySelector('h1').textContent).toStrictEqual('Oh no!');
  });
});
