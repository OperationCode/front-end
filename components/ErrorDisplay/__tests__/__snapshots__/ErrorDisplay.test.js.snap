// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDisplay should render with just required props 1`] = `
<div
  className="ErrorDisplay"
>
  <div
    className="bg"
  >
    <div
      className="textContainer"
    >
      <h1
        className="title"
      >
        404
        !
      </h1>
      <p
        className="paragraph"
      >
        We're so ashamed. You definitely weren't supposed to see this...
      </p>
    </div>
  </div>
</div>
`;
